runtime/lua/vim/lsp/rpc.lua

```diff
@@ -262,15 +286,18 @@ function Client:encode_and_send(payload)
   if self.transport.is_closing() then
     return false
   end
-  local encoded = vim.json.encode(payload)
+  local encoded = assert(
+    vim.json.encode(payload),
+    string.format("Couldn't encode payload '%s'", vim.inspect(payload))
+  )
   self.transport.write(format_message_with_content_length(encoded))
   return true
 end
 
 ---@private
 --- Sends a notification to the LSP server.
 ---@param method (string) The invoked LSP method
----@param params (any): Parameters for the invoked LSP method
+---@param params (any) Parameters for the invoked LSP method
```

Comment:
In other places `params` is of type `table?`. Is there a reason why we chose `any` here?

Reply:
This seems to have been changed on #21753 because `vim.json.encode` receives `any` as a parameter. Should other usages of `params` be updated?

Reply:
I see. Neh, leave it as it. Don't want to add noisy changes to this PR.